scripts/config.pl

```diff
@@ -218,7 +223,7 @@
 
 my $done;
 for my $line (@config_lines) {
-    if ($action eq "full" || $action eq "realfull" || $action eq "baremetal" ) {
+    if (grep {$action eq $_} qw(docfull full realfull baremetal)) {
         if ($line =~ /name SECTION: Module configuration options/) {
```

Comment:
Note that including “Module configuration options” when generating the documentation fixes the bug that some configuration options are missing (https://github.com/ARMmbed/mbedtls/issues/520), but introduces a minor bug: many options have lines like `//#define MBEDTLS_FOO 42 /**< Description */` that were copied from another header with `//` prepended, and this results in a duplicated sentence in the rendered documentation since Doxygen concatenates the documentation blocks from `config.h` and the other header.

Reply:
Revisiting this four years later, it seems wrong. `#define MBEDTLS_FOO` in the other header must be guarded by `#if !defined(MBEDTLS_FOO)`, otherwise it would be impossible to set it through `config.h`. But that implies that Doxygen will only ever see one of the documentation strings.